### PR TITLE
ActorBenchmark ported to typed

### PR DESCRIPTION
**Pull request by johanandren**

### CODE REVIEW

**Comment by johanandren** [comment]
Results from running on my machine:

```
akka-bench-jmh/jmh:run -f 1 -wi 3 -i 5 -prof gc .*\.(Typed)ActorBenchmark.*

throughput
[info] # Run complete. Total time: 00:01:49
[info] Benchmark            (batchSize)    (dispatcher)                         (mailbox)  (tpt)   Mode  Cnt         Score         Error  Units
[info] ActorBenchmark.echo           50  fjp-dispatcher  akka.actor.ManyToOneArrayMailbox     50  thrpt    5  61448895.698 ± 7549827.882  ops/s
[info] TypedActorBenchmark.echo      50  fjp-dispatcher  akka.actor.ManyToOneArrayMailbox     50  thrpt    5  38688398.669 ±  279460.100  ops/s


-prof gc
[info] Benchmark                                                   (batchSize)    (dispatcher)                         (mailbox)  (tpt)   Mode  Cnt         Score        Error   Units
[info] ActorBenchmark.echo                                                 50  fjp-dispatcher  akka.actor.ManyToOneArrayMailbox     50  thrpt    5  64058898.046 ± 323545.630   ops/s
[info] ActorBenchmark.echo:·gc.alloc.rate                                  50  fjp-dispatcher  akka.actor.ManyToOneArrayMailbox     50  thrpt    5      1138.016 ±   2449.483  MB/sec
[info] ActorBenchmark.echo:·gc.alloc.rate.norm                             50  fjp-dispatcher  akka.actor.ManyToOneArrayMailbox     50  thrpt    5        19.211 ±     41.350    B/op
[info] ActorBenchmark.echo:·gc.churn.G1_Eden_Space                         50  fjp-dispatcher  akka.actor.ManyToOneArrayMailbox     50  thrpt    5      1427.622 ±    196.434  MB/sec
[info] ActorBenchmark.echo:·gc.churn.G1_Eden_Space.norm                    50  fjp-dispatcher  akka.actor.ManyToOneArrayMailbox     50  thrpt    5        24.109 ±      3.242    B/op
[info] ActorBenchmark.echo:·gc.churn.G1_Old_Gen                            50  fjp-dispatcher  akka.actor.ManyToOneArrayMailbox     50  thrpt    5         0.145 ±      0.208  MB/sec
[info] ActorBenchmark.echo:·gc.churn.G1_Old_Gen.norm                       50  fjp-dispatcher  akka.actor.ManyToOneArrayMailbox     50  thrpt    5         0.002 ±      0.004    B/op
[info] ActorBenchmark.echo:·gc.churn.G1_Survivor_Space                     50  fjp-dispatcher  akka.actor.ManyToOneArrayMailbox     50  thrpt    5         0.364 ±      0.165  MB/sec
[info] ActorBenchmark.echo:·gc.churn.G1_Survivor_Space.norm                50  fjp-dispatcher  akka.actor.ManyToOneArrayMailbox     50  thrpt    5         0.006 ±      0.003    B/op
[info] ActorBenchmark.echo:·gc.count                                       50  fjp-dispatcher  akka.actor.ManyToOneArrayMailbox     50  thrpt    5        64.000               counts
[info] ActorBenchmark.echo:·gc.time                                        50  fjp-dispatcher  akka.actor.ManyToOneArrayMailbox     50  thrpt    5       170.000                   ms

[info] TypedActorBenchmark.echo                                            50  fjp-dispatcher  akka.actor.ManyToOneArrayMailbox     50  thrpt    5  38767292.925 ± 6368732.413   ops/s
[info] TypedActorBenchmark.echo:·gc.alloc.rate                             50  fjp-dispatcher  akka.actor.ManyToOneArrayMailbox     50  thrpt    5      1148.987 ±    2483.467  MB/sec
[info] TypedActorBenchmark.echo:·gc.alloc.rate.norm                        50  fjp-dispatcher  akka.actor.ManyToOneArrayMailbox     50  thrpt    5        32.012 ±      68.909    B/op
[info] TypedActorBenchmark.echo:·gc.churn.G1_Eden_Space                    50  fjp-dispatcher  akka.actor.ManyToOneArrayMailbox     50  thrpt    5      1448.343 ±     233.351  MB/sec
[info] TypedActorBenchmark.echo:·gc.churn.G1_Eden_Space.norm               50  fjp-dispatcher  akka.actor.ManyToOneArrayMailbox     50  thrpt    5        40.178 ±       0.036    B/op
[info] TypedActorBenchmark.echo:·gc.churn.G1_Old_Gen                       50  fjp-dispatcher  akka.actor.ManyToOneArrayMailbox     50  thrpt    5         0.089 ±       0.058  MB/sec
[info] TypedActorBenchmark.echo:·gc.churn.G1_Old_Gen.norm                  50  fjp-dispatcher  akka.actor.ManyToOneArrayMailbox     50  thrpt    5         0.002 ±       0.002    B/op
[info] TypedActorBenchmark.echo:·gc.churn.G1_Survivor_Space                50  fjp-dispatcher  akka.actor.ManyToOneArrayMailbox     50  thrpt    5         0.395 ±       0.278  MB/sec
[info] TypedActorBenchmark.echo:·gc.churn.G1_Survivor_Space.norm           50  fjp-dispatcher  akka.actor.ManyToOneArrayMailbox     50  thrpt    5         0.011 ±       0.006    B/op
[info] TypedActorBenchmark.echo:·gc.count                                  50  fjp-dispatcher  akka.actor.ManyToOneArrayMailbox     50  thrpt    5        80.000                counts
[info] TypedActorBenchmark.echo:·gc.time                                   50  fjp-dispatcher  akka.actor.ManyToOneArrayMailbox     50  thrpt    5       233.000                    ms
```

File: akka-bench-jmh/src/main/scala/akka/actor/typed/TypedBenchmarkActors.scala [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+  case class Start(respondTo: ActorRef[Completed])
+  case class Completed(startNanoTime: Long)
+
+  def echoActorsSupervisor(numMessagesPerActorPair: Int, numActors: Int, dispatcher: String, batchSize: Int,
+                           shutdownTimeout: FiniteDuration): Behavior[Start] =
+    Behaviors.receive { (ctx, msg) ⇒
+      msg match {
+        case Start(respondTo) ⇒
+          // note: no protection against accidentally running bench sessions in paralell
```
Comment (chbatey) [inline akka-bench-jmh/src/main/scala/akka/actor/typed/TypedBenchmarkActors.scala]: parallel

File: akka-bench-jmh/src/main/scala/akka/actor/typed/TypedBenchmarkActors.scala [hunk 2]
```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed
+
+import java.util.concurrent.{ CountDownLatch, TimeUnit }
```
Comment (chbatey) [inline akka-bench-jmh/src/main/scala/akka/actor/typed/TypedBenchmarkActors.scala]: Clean up imports

File: akka-bench-jmh/src/main/scala/akka/actor/typed/TypedBenchmarkActors.scala [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            Behaviors.stopped
+          } else {
+            Behaviors.same
+          }
+
+      }
+    }.narrow[Unit]
+  }
+
+  private def printProgress(totalMessages: Long, numActors: Int, startNanoTime: Long) = {
```
Comment (chbatey) [inline akka-bench-jmh/src/main/scala/akka/actor/typed/TypedBenchmarkActors.scala]: Use existing one in BenchmarkActors?

**Review by chbatey — APPROVED** [review]
LGTM, i added a couple of comments you can ignore if you wish!

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**